src/config.h

```diff
@@ -55,11 +55,17 @@ struct enftun_config
 
     int xtt_enable;
     const char* xtt_remote_port;
-    const char* xtt_tcti;
-    const char* xtt_device;
-    const char* xtt_socket_host;
-    const char* xtt_socket_port;
     const char* xtt_basename;
+
+    int tpm_enable;
+    const char* tpm_tcti;
+    const char* tpm_device;
+    const char* tpm_socket_host;
+    const char* tpm_socket_port;
+    int tpm_hierarchy;
+    const char* tpm_password;
+    size_t tpm_password_len;
```

Comment:
`tpm_password` is a NULL-terminated string (like `tpm_tcti`, `tpm_device`, etc.) so explicitly store the length here?

Is it to support a password that includes a NULL character?  I'm not sure if
```
config_lookup_string(cfg, "identity.tpm.password",
                             &config->tpm_password);
```
handles a NULL escape sequence (`\x00`) correctly.

Reply:
This is because `xaptum-tpm` requires the explicit length parameter (i.e. doesn't assume NULL-terminated).

Note, though, that the `tpm_password_len` is just immediately derived as `strlen(tpm_password)`. So the `tpm_password_len` is not something that appears in the config file.

The reason `xaptum-tpm` requires the length is
1) To support NULLs in a password (there are references to using hash outputs as "passwords", e.g. for the platform authorization)
2) Based on the general philosophy in our crypto libraries of not assuming NULL-terminated strings but instead always taking buffer-and-length

That said, we can store *only* the `tpm_password` in the config, and calculate `strlen(tpm_password)` when we pass it to the underlying functions (the `xaptum-tpm` initialization, as well as the `xtt` initialization). Other than potentially repeating the `strlen` calculation, it's six-of-one to me, so if you want me to clean up the config that's fine with me

Reply:
> That said, we can store _only_ the `tpm_password` in the config, and calculate `strlen(tpm_password)` when we pass it to the underlying functions (the `xaptum-tpm` initialization, as well as the `xtt` initialization).

Yeah, let's do that.     It will make it most obvious that `config.tpm_password` is standard c string (NULL terminated) and be consistent with how all the other config strings are handled.

Ultimately I'd like to refactor a bunch of the config stuff (the argument lists for the XTT and TLS functions are getting ridiculous!), but this isn't the PR to do it in.